scraper/spiders/citycouncil.py

```diff
@@ -75,98 +74,112 @@ def parse_page(self, response):
-                date_obj = parse(list_date, dayfirst=True)
-                if date_obj.date() >= self.start_date:
-                    found = True
-                    list_url = box.css("div a::attr(href)").get()
-                    yield scrapy.Request(
-                        url=response.urljoin(list_url),
-                        callback=self.parse_list_page,
-                        meta={"date": list_date},
-                    )
-
-        if found and last_page:  # deve continuar checando até não encontrar mais
-            last_page = int(last_page)
-            next_page = current_page + 1
-            if next_page <= last_page:
-                yield scrapy.Request(
-                    url=response.urljoin(f"listadepresenca.asp?p={next_page}"),
-                    callback=self.parse,
-                    meta={"current_page": next_page},
+                yield scrapy.FormRequest(
+                    url,
+                    formdata=data,
+                    callback=self.parse_list_page,
+                    meta={"event": event},
                 )
 
     def parse_list_page(self, response):
-        council_members = response.css("div.row div div ul li a::text").extract()
-        status = response.css("div.row div div div a::text").extract()
+        dates = response.css("div#myTabContent2 h4::text").extract()
+        dates = self.remove_invalid_dates(dates)
+
+        tables = response.css("table.table")
+
+        for a_date in dates:
+            table = tables.pop()
+            info = table.css("div.TITULO::text").extract()
+            council_members = []
+            status = []
+            index = 0
+            while index < len(info):
+                if index % 2 == 0:  # even
+                    council_members.append(info[index])
+                else:
+                    status.append(info[index])
+                index += 1
+
+            for council_member, status in zip(council_members, status):
+                yield CityCouncilAttendanceListItem(
+                    crawled_at=datetime_utcnow_aware(),
+                    crawled_from=response.url,
+                    date=from_str_to_date(a_date),
+                    council_member=council_member.strip(),
+                    status=get_status(status),
+                    description=response.meta["event"],
+                )
 
-        for council_member, status in zip(council_members, status):
-            yield CityCouncilAttendanceListItem(
-                crawled_at=datetime_utcnow_aware(),
-                crawled_from=response.url,
-                date=from_str_to_date(response.meta["date"]),
-                council_member=council_member.strip(),
-                status=self.get_status(status),
-            )
+    @staticmethod
+    def remove_invalid_dates(dates):
+        for c_date in range(0, len(dates)):
+            if c_date >= len(dates):
+                break
+
+            if "      " in dates[c_date]:
```

Comment:
Já que você quer iterar sobre os valores, basta percorrer as datas com um `for` que acessa essas datas. Além disso, se um espaço a mais for adicionado essa condição vai falhar. Então é melhor dar um `strip` na data pra eliminar todos os espaços em branco do início e do final; quando estiver vazio, aí vai virar uma string vazia. Então podemos ter algo como:

```
cleaned_dates = []
for a_date in dates:
    if a_date.strip() != "":
        cleaned_dates.append(a_date.strip())
return cleaned_dates
```

Aqui estou criando uma nova variável porque é uma boa prática não manipular a lista que estamos percorrendo porque pode causar um problema na lógica por causa de como as informações internas do Python estão organizadas.

Como essa função tem um papel importante (que adiciona um dado ou não), ela precisa ter testes. Você pode tirá-la do spider e colocar com outras funções utilitárias. Uma outra alternativa também e colocar dentro do `for a_date in dates:` no `def parse_list_page(self, response):` esse `if` que tem aqui. Se não for vazia, continua o processo. Se for vazia, `continue` pro próximo.

Alguns exemplos que consegui extrair e podem ser úteis pra escrever testes:

```
>>>>>>>>>>>>>>>>>> []
>>>>>>>>>>>>>>>>>> ['19/01/2023']
>>>>>>>>>>>>>>>>>> ['02/02/2023', '07/02/2023', '09/02/2023', '14/02/2023', '16/02/2023', '23/02/2023']
>>>>>>>>>>>>>>>>>> ['02/05/2023', '03/05/2023', '04/05/2023', '09/05/2023', '10/05/2023', '11/05/2023', '16/05/2023', '        ', '17/05/2023', '        ', '18/05/2023', '        \r\n        ', '23/05/2023', '        ']
>>>>>>>>>>>>>>>>>> []
```

Reply:
Preferi colocar o continue, para nao precisar criar uma nova funcao que algo basico resolve!